### PR TITLE
JSONReader load_data metadata

### PR DESCRIPTION
**Pull request by thegunslingers**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

The JSONReader's load_data function was updated to accept metadata. The motivation for this PR is to resolve an bug that I ran into.

Without this update, an error occurs when loading json files via the SimpleDirectoryReader along with the JSONReader. The error causes the following message: "Failed to load file file/path/name.json with error: load_data() got an unexpected keyword argument 'extra_info'. Skipping..."

Context:
```
documents = SimpleDirectoryReader(doc_dir, 
                                                file_extractor={'.json': JSONReader()}, 
                                                required_exts=[".json"], recursive=False).load_data()
```
This code results in the following:
llama_index/core/readers/file/base.py
Line 379 creates extra_info
Line 382 calls JSONReader with extra_info and causes error

Fixes # (issue) N/A

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [x] These changes were required to get my code to work. Developed, tested via debugger.
- [x] Ran `Make test` for llama-index-integrations/readers/llama-index-readers-json & llama-index-core/tests/readers/test_json.py

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-legacy/llama_index/legacy/readers/json.py [hunk 1]
```diff
@@ -84,7 +84,9 @@ def __init__(
         self.ensure_ascii = ensure_ascii
         self.is_jsonl = is_jsonl
 
-    def load_data(self, input_file: str) -> List[Document]:
+    def load_data(
```
Comment (hatianzhang) [inline llama-index-legacy/llama_index/legacy/readers/json.py:87]: @thegunslingers could you revert the changes to the legacy folder.
the legacy folder will be deprecated

Comment (thegunslingers) [reply llama-index-legacy/llama_index/legacy/readers/json.py]: Reverted changes to legacy JSONReader

**Review by hatianzhang — APPROVED** [review]
